src/explore-education-statistics-frontend/src/modules/table-tool/components/TimePeriodDataTable.tsx

```diff
@@ -24,156 +24,167 @@ interface Props {
   locations: LocationFilter[];
   results: TableData['result'];
   footnotes?: TableData['footnotes'];
+  tableHeadersConfig?: TableHeadersFormValues;
 }
 
-const TimePeriodDataTable = (props: Props) => {
-  const {
-    filters,
-    timePeriods,
-    locations,
-    indicators,
-    results,
-    footnotes,
-  } = props;
-
-  const dataTableRef = useRef<HTMLTableElement>(null);
-
-  const [tableHeaders, setTableHeaders] = useState<TableHeadersFormValues>({
-    columnGroups: [],
-    columns: [],
-    rowGroups: [],
-    rows: [],
-  });
-
-  const removeSiblinglessTotalRows = (
-    categoryFilters: Dictionary<CategoryFilter[]>,
-  ): CategoryFilter[][] => {
-    return Object.values(categoryFilters).filter(filter => {
-      return filter.length > 1 || !filter[0].isTotal;
+const TimePeriodDataTable = forwardRef<HTMLElement, Props>(
+  (props: Props, dataTableRef) => {
+    const {
+      filters,
+      timePeriods,
+      locations,
+      indicators,
+      results,
+      footnotes,
+      tableHeadersConfig,
+    } = props;
+
+    const [tableHeaders, setTableHeaders] = useState<TableHeadersFormValues>({
+      columnGroups:
+        tableHeadersConfig && tableHeadersConfig.columnGroups
+          ? tableHeadersConfig.columnGroups
+          : [],
+      columns:
+        tableHeadersConfig && tableHeadersConfig.columns
+          ? tableHeadersConfig.columns
+          : [],
+      rowGroups:
+        tableHeadersConfig && tableHeadersConfig.rowGroups
+          ? tableHeadersConfig.rowGroups
+          : [],
+      rows:
+        tableHeadersConfig && tableHeadersConfig.rows
+          ? tableHeadersConfig.rows
+          : [],
+    });
+    useEffect(() => {
```

Comment:
This `useEffect` should be redundant if we get rid of `tableHeaders` as state.

On a side-note, it's probably not a good idea to operate on the same state in two different `useEffect` blocks as we don't want to look in multiple places where the state might be getting updated.